src/mlpack/methods/ann/layer/nearest_interpolation_impl.hpp

```diff
@@ -0,0 +1,162 @@
+/**
+ * @file methods/ann/layer/nearest_interpolation_impl.hpp
+ * @author Abhinav Anand
+ *
+ * Implementation of the NearestInterpolation layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_NEAREST_INTERPOLATION_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_NEAREST_INTERPOLATION_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "nearest_interpolation.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+
+template<typename InputDataType, typename OutputDataType>
+NearestInterpolation<InputDataType, OutputDataType>::
+  NearestInterpolation():
+  inRowSize(0),
+  inColSize(0),
+  outRowSize(0),
+  outColSize(0),
+  depth(0),
+  batchSize(0)
+  {
+    // Nothing to do here.
+  }
+
+template<typename InputDataType, typename OutputDataType>
+NearestInterpolation<InputDataType, OutputDataType>::
+  NearestInterpolation(
+    const size_t inRowSize,
+    const size_t inColSize,
+    const size_t outRowSize,
+    const size_t outColSize,
+    const size_t depth):
+  inRowSize(inRowSize),
+  inColSize(inColSize),
+  outRowSize(outRowSize),
+  outColSize(outColSize),
+  depth(depth),
+  batchSize(0)
+  {
+    // Nothing to do here.
+  }
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void NearestInterpolation<InputDataType, OutputDataType>::Forward(
+  const arma::Mat<eT>& input, arma::Mat<eT>& output)
+  {
+    batchSize = input.n_cols;
+    if (output.is_empty())
+      output.set_size(outRowSize * outColSize * depth, batchSize);
+    else
+    {
+      assert(output.n_rows == outRowSize * outColSize * depth);
+      assert(output.n_cols == batchSize);
+    }
+
+    assert(inRowSize >= 2);
+    assert(inColSize >= 2);
+
+    arma::cube inputAsCube(const_cast<arma::Mat<eT>&>(input).memptr(),
+      inRowSize, inColSize, depth * batchSize, false, false);
+    arma::cube outputAsCube(output.memptr(), outRowSize, outColSize,
+      depth * batchSize, false, true);
```

Comment:
```suggestion
        inRowSize, inColSize, depth * batchSize, false, false);
    arma::cube outputAsCube(output.memptr(), outRowSize, outColSize,
        depth * batchSize, false, true);
```

Tab twice, see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping for more details.